design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,129 @@
+1. An app Pod may then mount the Secret (annotated in the config map)  and the ConfigMap keys to begin accessing the bucket. 
+
+**Use Case:** Delete an Object Bucket 
+_As a Kubernetes user, I want to use kubectl to delete object bucket API resources owned by a given object user. (Note: the underlying S3 bucket is not deleted. Permanent deletion is left to the admin.)_
+
+1. The user deletes the CephObjectBucket.  This triggers an automated cleanup sequence:
+1. The CephObjectBucket is marked for deletion first and left in the foreground. Deletion is blocked by a finalizer on the CephObjectBucket, pending step (b).
+1. The respective ConfigMap is deleted.
+1. The finalizer in the CephObjectBucket is removed and the object is garbage collected.
+
+## API Specifications
+
+### CephObjectBucket
+
+```yaml
+apiVersion: object.k8s.io/v1alpha
+kind: CephObjectBucket
+metadata:
+  name: my-bucket-1
+  namespace: dev-user
+  annotations:     # added by provisioner
+    bucket_owner: “dev-user/my-user-1”
+  finalizers:      # added by provisioner
+  - rook-ceph.io/provisioner/bucket
+  ownerReferences: # added by provisioner
+  - apiVersion: ceph.rook.io/v1beta1
+    blockOwnerDeletion: true
+    kind: ObjectStoreUser
+    name: my-user-1
+    uid: 1234-3456-7839-1231
+spec:
+  bucketName:<string>         [1]
+  generateBucketName:<string> [1]
+  ownerUser: my-user # required, name of object user’s metadata.Name
+status:
+  phase: [ “creating”, “available”, ...]
+
+```
+
+[1] `bucketName` and `generateBucketName` are mutually exclusive. It is expected that `generateBucketName` be the common usage. This field produces a random bucket name beginning with the value of this stanza. If bucketName is defined then that value alone is used as the name of the bucket. Within an object store, buckets must be unique, so if bucketName is used there may be a higher chance of name collision. The api will continue retrying if the bucket cannot be created.
+
+### CephObjectBucketConfigMap
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rook-ceph-object-bucket-my-bucket-1
+  namespace: dev-user
+  annotations:
+    bucket_secret: “rook-ceph-object-user-my-user-1”
+    bucket_owner: “my-user-1”
+  ownerReferences:
+  - name: my-bucket-1
+    uid: 1234-qwer-4321-rewq
+    apiVersion: object.k8s.io/v1alpha
+    kind: CephObjectBucket 
+  finalizers:
+  - rook-ceph.io/provisioner/my-bucket-1
+data:
+  BUCKET_HOST: http://my-store-url-or-ip
+  BUCKET_PORT: 80
+  BUCKET_NAME: my-bucket-1
+  BUCKET_SSL: no
+```
+
+### objectStorageClass
+
+```yaml
+apiVersion: storage.k8s.io/v1
+   kind: StorageClass
+   metadata:
+     name: rook-ceph-object-store
+   provisioner: kubernetes.io/rook-ceph-operator
+   parameters:
+     # namespace/objectStoreService
+     objectStore: rook-ceph.io/objectStoreName
```

Comment:
> Instead of namespace/objectStoreName, why not have two separate properties for namespace and objectStore?

I'm fine with that, will change.

> was rook-ceph.io intended for the namespace or should this be rook-ceph?
nope! that's a typo, will fix